utils/HtmlElems.ts

```diff
@@ -0,0 +1,183 @@
+    let style: string
+    if ( clickedBtn === CurrentButton.Remove ) {
+        style = "opacity: 100%"
+    } else {
+        style = "opacity:0.5"
+    }
+    return `<a id="removeBtn"
+    class="waves-effect  waves-light btn" style=${style}>
+    Remove
+</a>`
+}
+export const ShowAllBtn = ( clickedBtn: CurrentButton ) => {
+    let style: string
+    if ( clickedBtn === CurrentButton.ShowAll ) {
+        style = "opacity: 100%"
+    } else {
+        style = "opacity:0.5"
+    }
+    return `<a id="showAllBtn"
+    class="waves-effect  waves-light btn" style=${style}>
+    Show All
+</a>`
+
+}
+
+
+// Submit button as html
+export const SubmBtn = ( clickedBtn: CurrentButton ) => {
+    let visability: string
+    let btnName: string
+    if ( clickedBtn === CurrentButton.Search ) {
+        visability = "block"
+        btnName = "Search"
+    }
+    if ( clickedBtn === CurrentButton.Add ) {
+        visability = "block"
+        btnName = "Add"
+    }
+    if ( clickedBtn === CurrentButton.Remove ) {
+        visability = "block"
+        btnName = "Remove"
+    }
+    if ( clickedBtn === CurrentButton.ShowAll )
+        visability = "none"
+
+    return `<a id="submitBtn" ${status}
+    class="waves-effect  waves-light btn red" style="display: ${visability};" >
+   ${btnName}
+    </a>`
+
+}
+
+
+// Space between elements 
+export const Space = () => `<br>
+                            <br>
+                            <br>`
+
+// Hidden by default div 
+export const HiddenDiv = ( clickedBtn: CurrentButton, users: Array<User>, ifSbmtClicked: boolean, ...content: Array<string> ) => {
+    let status: string
+    if ( clickedBtn === CurrentButton.Search )
+        status = "hidden"
+    if ( clickedBtn === CurrentButton.Search, users.length > 0, ifSbmtClicked === true )
+        status = "block"
+    if ( clickedBtn === CurrentButton.Add )
+        status = "hidden"
+    if ( clickedBtn === CurrentButton.Remove )
+        status = "hidden"
+    if ( clickedBtn === CurrentButton.ShowAll )
+        status = "block"
+    return `<div id="hiddenDiv" ${status}>${content.join( "" )}</div>`
+}
+
+
+// Table
+export const TableElem = ( ...content: Array<string> ) => `<table id="UTable" 
+class="responsive-table responsive-table col s4 offset-s4">
+                                                    ${content.join( "" )}
+                                                </table>`
+
+// Head element of the table
+export const Thead = ( ...content: Array<string> ) => `<thead>
+                                    <tr>
+                                        ${content.join( "" )}
+                                    </tr>
+                                </thead>`
+
+// Name the category of table
+export const TheadName = ( ...content: Array<string> ) => `<th class="col s3">${content}</th>`
+
+// Table element with users information
+export const Tbody = ( users: Array<User> ) => {
+
+    let usersAsContent: Array<string> = []
```

Comment:
Why `let`? You are pushing to the array in the code below, but not replacing the array.